src/runs.tsx

```diff
@@ -23,34 +41,49 @@ type RouteParams = { id: string };
 
 interface Props extends RouteComponentProps<RouteParams> { }
 
-class RunsPage extends React.Component<Props, IState> {
-    public constructor(props: Props) {
+class RunsPage extends React.Component<Props & IProps, IState> {
+    public constructor(props: Props & IProps) {
         super(props);
         this.state = {
             catalogs: []
         };
     }
 
     render() {
+        const { classes } = this.props;
         return (
-          <Container maxWidth="sm">
-          <Box my={4}>
-            <Typography variant="h4" component="h1" gutterBottom>
-              This page lists runs for {this.props.match.params.id}
-              </Typography>
-              <Typography variant="h6" component="h1" gutterBottom>
-              <div>
-                <ul>
-                    {this.state.catalogs.map(catalog => (
-                        <li key={catalog.uid}>
-                            <Link color="inherit" component={RouterLink} to={"/runs/" + this.props.match.params.id + "/" + catalog.uid}>{catalog.uid}</Link>
-                        </li>
-                    ))}
-                </ul>
-              </div>
-            </Typography>
+          <div style={{ width: '100%' }}>
+              Showing {this.props.match.params.id}
+          <Box display="flex" flexWrap="wrap" p={1} m={1} bgcolor="background.paper">
+          {this.state.catalogs.map(catalog => (
+          <Card className={classes.root}>
+            <CardActionArea>
+              <CardMedia
+                className={classes.media}
+                image={"/previews/" + catalog.uid}
+                title="Run preview"
+              />
+              <CardContent>
+                <Typography gutterBottom variant="h5" component="h3">
+                  {catalog.uid}
+                </Typography>
+                <Typography variant="body2" color="textSecondary" component="p">
+                  Maybe scan notes, summary of the scan?
+                </Typography>
+              </CardContent>
+            </CardActionArea>
+            <CardActions>
+              <Button size="small" color="primary" >
+                Share
+              </Button>
+              <Button size="small" color="primary" component={RouterLink} to={"/runs/" + this.props.match.params.id + "/" + catalog.uid}>
```

Comment:
Is there something like Python's f-strings to format it easier?
```suggestion
              <Button size="small" color="primary" component={RouterLink} to={"/db/runs/" + this.props.match.params.id + "/" + catalog.uid}>
```

Reply:
AFAIK the answer is sadly no.

Reply:
OK, no worries. I am just curious if `this.props.match.params.id` is of a wrong type, would it concatenate properly? looks like JavaScript is pretty forgiving, but maybe not TypeScript.